Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -292,7 +292,15 @@ def bottle_formula(f)
 
         cd cellar do
           sudo_purge
-          safe_system "tar", "cf", tar_path, "#{f.name}/#{f.pkg_version}"
+          keg_root = "#{f.name}/#{f.pkg_version}"
+          safe_system "tar", "cf", tar_path,
+                      # Put the two metafiles at the start of the tarball,
+                      # before listing the actual root of the keg.
+                      # This ensures that even a partial tarball download
+                      # should contain these two files.
+                      "#{keg_root}/INSTALL_RECEIPT.json",
+                      "#{keg_root}/.brew/#{f.name}.rb",
```

Comment:
I wonder if it's worth having this be a keg method? Could be used in `keg_relocate.rb` and `utils/bottles.rb`

Reply:
One of these paths is absolute and one is relative. I could do a simple helper like this, though:

```diff
diff --git a/Library/Homebrew/dev-cmd/bottle.rb b/Library/Homebrew/dev-cmd/bottle.rb
index 37713fdf7f..5de7567e83 100644
--- a/Library/Homebrew/dev-cmd/bottle.rb
+++ b/Library/Homebrew/dev-cmd/bottle.rb
@@ -299,7 +299,7 @@ module Homebrew
                       # This ensures that even a partial tarball download
                       # should contain these two files.
                       "#{keg_root}/INSTALL_RECEIPT.json",
-                      "#{keg_root}/.brew/#{f.name}.rb",
+                      "#{keg_root}/#{keg.relative_formula_path}",
                       keg_root
           sudo_purge
           tar_path.utime(tab.source_modified_time, tab.source_modified_time)
diff --git a/Library/Homebrew/keg.rb b/Library/Homebrew/keg.rb
index 6414640bf4..9e5966be73 100644
--- a/Library/Homebrew/keg.rb
+++ b/Library/Homebrew/keg.rb
@@ -549,6 +549,10 @@ class Keg
     find { |pn| FileUtils.rm_rf pn if pn.basename.to_s == "__pycache__" }
   end
 
+  def relative_formula_path
+    ".brew/#{name}.rb"
+  end
+
   private
 
   def resolve_any_conflicts(dst, mode)
diff --git a/Library/Homebrew/keg_relocate.rb b/Library/Homebrew/keg_relocate.rb
index b8028ceab5..316142795c 100644
--- a/Library/Homebrew/keg_relocate.rb
+++ b/Library/Homebrew/keg_relocate.rb
@@ -130,7 +130,7 @@ class Keg
         next true if pn.symlink?
         next true if pn.directory?
         next false if pn.basename.to_s == "orig-prefix.txt" # for python virtualenvs
-        next true if pn == self/".brew/#{name}.rb"
+        next true if pn == self/relative_formula_path
         next true if Metafiles::EXTENSIONS.include?(pn.extname)
 
         if pn.text_executable?
```

Reply:
Sounds good 👍